R/misc.R

```diff
@@ -155,3 +155,105 @@ vectorOrList <- function (obj, type) {
     }
     return(FALSE)
 }
+
+#' Grab either env variable or .Rprofile based option
+#' 
+#' .Rprofile options are like "test.api", while env vars are "R_TEST_API", 
+#'  this function will grab the environment variable if it is find, otherwise
+#'  it looks for the R-based option value.
+#' 
+#' @param opt the option to get
+#' 
+#' @return the value of the option
+#' 
+#' @export
+envOrOption <- function (opt) {
+    ## .Rprofile options are like "test.api", while env vars are "R_TEST_API"
+    envvar.name <- paste0("R_", toupper(gsub(".", "_", opt, fixed=TRUE)))
+    envvar <- Sys.getenv(envvar.name)
+    if (nchar(envvar)) {
+        ## Let environment variable override .Rprofile, if defined
+        return(envvar)
+    } else {
+        return(getOption(opt))
+    }
+}
+
+
+#' Grab either env variable or .Rprofile based option
+#' 
+#' .Rprofile options are like "test.api", while env vars are "R_TEST_API", 
+#'  this function will grab the environment variable if it is find, otherwise
+#'  it looks for the R-based option value.
+#' 
+#' @param opt the option to get
+#' 
+#' @return the value of the option
+#' 
+#' @keywords internal
+#' @export
+envOrOption <- function (opt) {
+    ## .Rprofile options are like "test.api", while env vars are "R_TEST_API"
+    envvar.name <- paste0("R_", toupper(gsub(".", "_", opt, fixed=TRUE)))
+    envvar <- Sys.getenv(envvar.name)
+    if (nchar(envvar)) {
+        ## Let environment variable override .Rprofile, if defined
+        return(envvar)
+    } else {
+        return(getOption(opt))
+    }
+}
+
+#' Change which server to point to
+#' 
+#' A convenience function for changing where you want the Crunch package to try
+#' to connect to. You can select either:
+#' 
+#' * 'local' to connect to a local (vagrant-based) backend. This will use the 
+#' `test.user`, `test.pw`, and `test.api` options from your `.Rprofile` or 
+#' corresponding environment variables.
+#' 
+#' * 'prod' to connect to a hosted backend. This will use the `crunch.email`,
+#' `crunch.pw`, and `crunch.api` options from your `.Rprofile` or corresponding
+#'  environment variables.
+#' 
+#' @param place a place to point to (either 'local' or 'prod')
+#' @param api_url a url to use by default: "https://app.crunch.io/api/"
+#' @param use_env_api_url should the api url from the environment variables be
+#' used? (default: `FALSE`, this will always be `FALSE` when choosing `prod = 'local'`)
+#' 
+#' @return nothing
+#' 
+#' @keywords internal
+#' @export
+pointTo <- function (place, api_url = "https://app.crunch.io/api/", use_env_api_url = FALSE) {
```

Comment:
I don't think this function belongs in the package. I could see something like:

```
setCrunchAPI <- function (subdomain, port=NULL) {
    if (!is.null(port)) {
        api <- paste0("http://", subdomain, ".crunch.io:", port, "/api/")
    } else {
        api <- paste0("https://", subdomain", .crunch.io/api/")
    options(crunch.api=api)
}
```

and you could then have `login` use `envOrOption` for email and pw (see https://github.com/Crunch-io/rcrunch/blob/master/R/auth.R#L50). 

With those tools, could you build in the other ad hoc behaviors to `crunchdev`?